src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -198,6 +208,54 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
             executionContext.Debug("Diagnostic file upload complete.");
         }
 
+        // Dumping Agent Azure VM extension logs to the support files folder.
+        private void DumpAgentExtensionLogs(IExecutionContext executionContext, string supportFilesFolder, DateTime jobStartTimeUtc)
+        {
+            executionContext.Debug("Starting dumping agent extension logs.");
+
+            string pathToLogs = null;
+            string archiveName = null;
+            string timestamp = jobStartTimeUtc.ToString("yyyyMMdd-HHmmss");
+
+            if (PlatformUtil.RunningOnWindows)
+            {
+                string pathToVersions = "C:\\WindowsAzure\\Logs\\Plugins\\Microsoft.VisualStudio.Services.TeamServicesAgent";
+                if (Directory.Exists(pathToVersions))
+                {
+                    string[] subDirs = Directory.GetDirectories(pathToVersions).Select(dir => dir.Split("\\").Last()).ToArray();
+                    Version[] versions = subDirs.Select(dir => new Version(dir)).ToArray();
+                    Version maxVersion = versions.Max();
+                    pathToLogs = Path.Combine(pathToVersions, maxVersion.ToString());
+                }
+                archiveName = $"AgentWindowsExtensionLogs-{timestamp}-utc.zip";
+            }
+
+            if (PlatformUtil.RunningOnLinux)
+            {
+                pathToLogs = "/var/log/azure/Microsoft.VisualStudio.Services.TeamServicesAgentLinux";
+                archiveName = $"AgentLinuxExtensionLogs-{timestamp}-utc.zip";
+            }
+
+            if (Directory.Exists(pathToLogs))
```

Comment:
Instead of several checks on folder existence could it make more sense to check that the target folder exists only one time at the beginning method work?

Reply:
Refactored so that the `Directory.Exists` method is used once
(either in the `PlatformUtil.RunningOnWindows` case or in the `PlatformUtil.RunningOnLinux` case).